linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pwquality/accounts_password_pam_retry/bash/shared.sh

```diff
@@ -1,9 +1,34 @@
 # platform = multi_platform_all
 
+{{% if product in ['ol8','ol9','rhel8', 'rhel9'] %}}
+{{% set configuration_files = ["password-auth","system-auth"] %}}
+{{% else %}}
+{{% set configuration_files = ["system-auth"] %}}
+{{% endif %}}
+
+
 {{{ bash_instantiate_variables("var_password_pam_retry") }}}
 
-{{% if product in ['rhel8', 'rhel9'] -%}}
-	{{{ bash_replace_or_append('/etc/security/pwquality.conf', '^retry', '$var_password_pam_retry', '%s = %s') }}}
+{{% if product in ['ol8','ol9','rhel8', 'rhel9'] -%}}
+	{{{ bash_replace_or_append('/etc/security/pwquality.conf',
+							   '^retry',
+							   '$var_password_pam_retry',
+							   '%s = %s') }}}
+	{{% for cfile in configuration_files %}}
+		{{{ bash_remove_pam_module_option_configuration(pam_file='/etc/pam.d/' ~ cfile,
+									  	  				group='password',
+														control=".*",
+									      				module='pam_pwquality.so',
+									      				option='retry') }}}
+	{{% endfor %}}
 {{% else %}}
-	{{{ bash_ensure_pam_module_option('/etc/pam.d/system-auth', 'password', 'requisite', 'pam_pwquality.so', 'retry', "$var_password_pam_retry", '^account.*required.*pam_permit.so') }}}
+	{{% for cfile in configuration_files %}}
+		{{{ bash_ensure_pam_module_configuration('/etc/pam.d/' ~ cfile,
+										  'password',
+										  'requisite',
+										  'pam_pwquality.so',
+										  'retry',
+										  "$var_password_pam_retry",
+										  '^\s*account') }}}
```

Comment:
The same comment from `shared.yml` is applicable here.

Reply:
I was having issues with OL7, so I updated the macro to only insert the new line after the last match, as it works in ansible

Reply:
I confirmed it is working fine now.